app/configurator/configurator-state.spec.tsx

```diff
@@ -1201,3 +1204,27 @@ describe("filtering", () => {
     ]);
   });
 });
+
+describe("publishing chart config", () => {
+  it("should run correct number of times in singleURLs layout mode", async () => {
+    const key = "ABC";
+    const publishableChartKeys = ["A", "C"];
+    const state = {
+      state: "PUBLISHING",
+      key,
+      layout: {
+        type: "singleURLs",
+        publishableChartKeys,
+      },
+      chartConfigs: [
+        { key: "A", cubes: [] },
+        { key: "B", cubes: [] },
+        { key: "C", cubes: [] },
+      ],
+    } as any as ConfiguratorStatePublishing;
+
+    const cb = jest.fn();
+    await publishState({} as any, key, state, async () => cb());
+    expect(cb).toHaveBeenCalledTimes(publishableChartKeys.length);
```

Comment:
Thanks for the test. Suggestion: it seems to me that the fix you introduced is not tested here ? Maybe we can add checks on the calls to make sure they use the proper i.

`expect(cb.calls.map(c => c.parameter[1]).toEqual([1, 2, 3])` ? (Not sure of the exact API of jest spies calls).

Reply:
Oh man 🤦‍♂️ Yes, you're right, I adjusted the test, thanks 😅